api/src/pcapi/routes/backoffice_v3/offerers.py

```diff
@@ -542,6 +568,77 @@ def toggle_top_actor(offerer_id: int) -> utils.BackofficeResponse:
     return _redirect_after_offerer_validation_action()
 
 
+def _offerer_batch_action(
+    api_function: typing.Callable,
+    success_message: str,
+    form: typing.Callable,
+) -> utils.BackofficeResponse:
+    try:
+        _form = form()
+        offerer_ids = [int(id) for id in _form.object_ids.data.split(",")]
+    except ValueError:
+        flash("Identifiant(s) invalide(s)")
+        return _redirect_after_offerer_validation_action(400)
+
+    offerers = offerers_models.Offerer.query.filter(offerers_models.Offerer.id.in_(offerer_ids)).all()
+
+    for offerer in offerers:
+        if hasattr(form, "tags"):
+            saved_homologation_tags = set(filter_homologation_tags(offerer.tags))
+            tags_to_add = set(_form.tags.data) - saved_homologation_tags
+            tags_to_remove = saved_homologation_tags - set(_form.tags.data)
+            api_function(
+                offerer, current_user, _form.comment.data, tags_to_add=tags_to_add, tags_to_remove=tags_to_remove
+            )
+        elif (
+            hasattr(form, "comment")
+            and isinstance(api_function, type(offerers_api.reject_offerer))
+            and not isinstance(api_function, type(offerers_api.validate_offerer))
+        ):
+            api_function(offerer, current_user, _form.comment.data)
+        else:
+            api_function(offerer, current_user)
```

Comment:
En fait je verrais bien un truc plus souple, du genre :
```
api_func_args = {"offerer": offerer}
if hasattr(form, "tags"):
    ...
    api_func_args.update({
        "tags_to_add": tags_to_add,
        "tags_to_remove": tags_to_remove,
    })
if hasattr(form, "comment"):
    api_func_args["comment"] = _form.comment.data
api_function(**args)
```

Pas testé, mais c'est l'idée

Reply:
la solution retenue :
"comment" in getfullargspec(api_function).args
ce qui suffit à différencier le reject du validate dans la fonction d'api offerer